### PR TITLE
Add support for state content customization in tile card

### PR DESCRIPTION
**Pull request by piitaya**
### Card

The state can be replaced by one or multiple attribute to avoid the usage of templating. `state` and `last-changed` key are allowed too. An option as been added to totally state the state.

CleanShot 2023-10-10 at 12 16 57

### Config

Two new options has been added.

```yaml
hide_state?: boolean
state_content?: string | string[]
```

### Editor

The editor uses a multiple select to let the user choose attributes. Items are draggable for re-ordering.

CleanShot 2023-10-16 at 14 47 44

### Demo

https://github.com/home-assistant/frontend/assets/5878303/8b0b1e13-60a9-42a3-b6c4-abec2533410b

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-selector/ha-selector-select.ts [hunk 1]
```diff
@@ -124,23 +186,25 @@ export class HaSelectSelector extends LitElement {
...
-                      @click=${this._removeItem}
-                    ></ha-svg-icon>
-                  </ha-chip>
-                `
-              )}
-            </ha-chip-set>`
-          : ""}
+          ? html`
+              <ha-chip-set>
+                ${value.map(
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-select.ts:191]: should this be `repeat` when we are going to sort them??

File: src/components/ha-selector/ha-selector-select.ts [hunk 2]
```diff
@@ -38,6 +42,64 @@ export class HaSelectSelector extends LitElement {
...
+      } else if (!this.value?.length && this._sortable) {
+        this._destroySortable();
+      }
+    }
+  }
+
+  private async _createSortable() {
+    const Sortable = await loadSortable();
+    this._sortable = new Sortable(
+      this.shadowRoot!.querySelector("ha-chip-set")!,
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-select.ts:60]: We should probably also add sort support to the checkbox option then?

Comment (piitaya) [reply src/components/ha-selector/ha-selector-select.ts:60]: I think we should force the dropdown layout if reorder selector option is true. We already do that if there is a custom value. Reordering checkbox is weird.

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-select.ts:65]: works for me

**Comment by piitaya** [comment]
@bramkragten I also added a `hide_state` option. We can also use `state_content: none` but having a `hide_state` boolean is easier to understand.

File: src/components/ha-selector/ha-selector-select.ts [hunk 3]
```diff
@@ -38,6 +43,68 @@ export class HaSelectSelector extends LitElement {
...
+
+  protected updated(changedProps: PropertyValues): void {
+    if (changedProps.has("value") || changedProps.has("selector")) {
+      if (
+        this.value?.length &&
+        !this._sortable &&
+        this.selector.select?.reorder
+      ) {
+        this._createSortable();
+      } else if (!this.value?.length && this._sortable) {
```
Comment (bramkragten) [inline src/components/ha-selector/ha-selector-select.ts:56]:
```suggestion
      } else if (this._sortable && (!this.selector.select?.reorder || !this.value?.length)) {
```

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**